libctru/source/system/syscalls.c

```diff
@@ -28,6 +31,64 @@ struct _reent* __SYSCALL(getreent)()
 	return tv->reent;
 }
 
+int __SYSCALL(clock_gettime)(clockid_t clock_id, struct timespec *tp) {
+
+	if (clock_id == CLOCK_REALTIME)
+	{
+		if (tp != NULL)
+		{
+			// Retrieve current time, adjusting epoch from 1900 to 1970
+			s64 ms_since_epoch = osGetTime() - 2208988800000ULL;
+			tp->tv_sec = ms_since_epoch / 1000;
+			tp->tv_nsec = (ms_since_epoch - 1000*tp->tv_sec) * 1000000;
+		}
+	}
+	else if (clock_id == CLOCK_MONOTONIC) 
+	{
+		if (tp != NULL)
+		{
+			// Use the ticks directly, as it offer the highest precision
+			u64 usec_since_boot = svcGetSystemTick() / CPU_TICKS_PER_USEC;
+
+			tp->tv_sec = usec_since_boot / 1000000;
+			tp->tv_nsec = (usec_since_boot - 1000000*tp->tv_sec) * 1000;
+		}
+	}
+	else
+	{
+		errno = EINVAL;
+		return -1;
+	}
+
+	return 0;
+}
+
+int __SYSCALL(clock_getres)(clockid_t clock_id, struct timespec *res) {
+	if (clock_id == CLOCK_REALTIME)
+	{
+		if (res != NULL) 
+		{
+			res->tv_sec = 0;
+			res->tv_nsec = 1000000;
+		}
+	}
+	else if (clock_id == CLOCK_MONOTONIC)
+	{
+		if (res != NULL)
+		{
+			res->tv_sec = 0;
+			res->tv_nsec = 1000;
```

Comment:
With the above change:
```cpp
res->tv_nsec = 1;
```

Reply:
The real resolution is a bit lower than that, but since it's not a whole number, I guess 1 is the best we can do?

Reply:
Yeah I just used 1 since we're on that order of magnitude and also because we could get every possible value of tv_nsec.